### PR TITLE
Fix behavior with @BeforeScenario and @AfterScenario

### PR DESCRIPTION
**Pull request by xsmrcek**

### CODE REVIEW

File: src/main/java/org/jbehavesupport/engine/reporter/StepLoggingReporter.java [hunk 1]
```diff
@@ -57,6 +58,10 @@ public class StepLoggingReporter extends AbstractLoggingReporter {
     private Deque<TestDescriptor> currentStepDescriptor = new ArrayDeque<>();
 
     private boolean isInBeforeStories = false;
+
+    private boolean isInBeforeScenario = false;
+
+    private boolean isInAfterScenario = false;
```
Comment (paveljandejsek) [inline src/main/java/org/jbehavesupport/engine/reporter/StepLoggingReporter.java]: ultra nitpick but I would remove the new lines on here 😄 
```suggestion
    private boolean isInBeforeStories = false;
    private boolean isInBeforeScenario = false;
    private boolean isInAfterScenario = false;
```

File: src/main/java/org/jbehavesupport/runner/reporter/JUnitStepReporter.java [hunk 2]
```diff
@@ -53,6 +54,10 @@ public class JUnitStepReporter extends AbstractJUnitReporter {
     private Deque<Description> currentStepDescription = new ArrayDeque<>();
 
     private boolean isInBeforeStories = false;
+
+    private boolean isInBeforeScenario = false;
+
+    private boolean isInAfterScenario = false;
```
Comment (paveljandejsek) [inline src/main/java/org/jbehavesupport/runner/reporter/JUnitStepReporter.java]: ultra nitpick #2
```suggestion
    private boolean isInBeforeStories = false;
    private boolean isInBeforeScenario = false;
    private boolean isInAfterScenario = false;
```

File: src/test/java/org/jbehavesupport/runner/story/steps/TestSteps.java [hunk 3]
```diff
@@ -34,10 +30,45 @@
...
+    public void afterStory() {
+        logger.info("After story custom step");
+    }
+
+    @AfterStories
+    public void afterStories() {
+        logger.info("After stories custom step");
+    }
+
     private static final Logger logger = LoggerFactory.getLogger(TestSteps.class);
```
Comment (paveljandejsek) [inline src/test/java/org/jbehavesupport/runner/story/steps/TestSteps.java]: Again nitpick, but could you move this field Logger logger to somewhere along line 30? And yes I know its my own mistake to put it in here 😄

File: src/test/java/org/jbehavesupport/runner/story/steps/TestSteps.java [hunk 4]
```diff
@@ -18,11 +18,7 @@
  */
 package org.jbehavesupport.runner.story.steps;
 
-import org.jbehave.core.annotations.BeforeStory;
-import org.jbehave.core.annotations.Composite;
-import org.jbehave.core.annotations.Given;
-import org.jbehave.core.annotations.Then;
-import org.jbehave.core.annotations.When;
+import org.jbehave.core.annotations.*;
```
Comment (paveljandejsek) [inline src/test/java/org/jbehavesupport/runner/story/steps/TestSteps.java]: I know its probably idea auto, but I would keep the single imports 🙂

**Review by paveljandejsek — APPROVED** [review]
Thanks a lot, superb work!

**Review by JakubVychodil — APPROVED** [review]

**Pull request merged**